index.d.ts

```diff
@@ -4,7 +4,7 @@ interface Options {
 
 	@default undefined
 	*/
-	readonly filter?: (fullPath: string) => boolean | undefined
+	readonly filter?: (fullPath: string) => boolean | void
```

Comment:
```suggestion
	readonly filter?: (fullPath: string) => boolean
```

The user should return a boolean.

Reply:
Since javascript does not enforce it, what do you want it to do if they return values outside of the range.  No other javascript function errors if not returning a boolean.

Reply:
the standard typing for filtering function that typescript uses is
`filter(callbackfn: (value: T, index: number, array: readonly T[]) => unknown, thisArg?: any): T[];`

Reply:
It's irrelevant what JS does already. The built-in TS types have to model exactly how JS works. We can make our code better and stricter.

Reply:
> Since javascript does not enforce it

It should